From e9e6a3e691893fd441fc12239cedb7cd728b63a9 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Tue, 18 Feb 2025 15:15:33 +0100
Subject: [PATCH 1540/1695] dts: stm32: expose vdda18adc regulator to CM33
 agent in stm32mp211.dtsi

Expose vdda18adc regulator to CM33 agent with ID 3.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9b92bd75e8cb4323ae350b9491790a40d53a592b dts: stm32: expose vdda18adc regulator to CM33 agent in stm32mp211.dtsi]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: Icd17227c3b0cfa22cf8a90379b4826aa479a5804
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438062
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi b/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
index 2f07e585c..6985f1d4f 100644
--- a/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
+++ b/core/arch/arm/dts/stm32mp21-st-scmi-cfg.dtsi
@@ -1557,6 +1557,11 @@
 				regulators {
 					#address-cells = <1>;
 					#size-cells = <0>;
+
+					regulator@3 {
+						reg = <VOLTD_SCMI_ADC>;
+						voltd-supply = <&vdda18adc>;
+					};
 				};
 			};
 		};
-- 
2.47.2

